UnityProject/Assets/Scripts/Objects/Botany/SeedExtractor.cs

```diff
@@ -139,39 +154,55 @@ public void DispenseSeedPacket(SeedPacket seedPacket)
-				{
-					Chat.AddActionMsgToChat(interaction.Performer,
-						$"You try and place the {foodAtributes.ArticleName} into the seed extractor but it is full!",
-						$"{interaction.Performer.name} tries to place the {foodAtributes.ArticleName} into the seed extractor but it is full!");
-					return;
+				AddSeedPacketToStorage(packet, interaction);
+				return;
 			}
+			var grownFood = interaction.HandObject.GetComponent<GrownFood>();
+			var foodAtributes = grownFood.GetComponentInParent<ItemAttributesV2>();
 
+			if (!Inventory.ServerTransfer(interaction.HandSlot, storage.GetBestSlotFor(interaction.HandObject)))
+			{
 				Chat.AddActionMsgToChat(interaction.Performer,
+					$"You try and place the {foodAtributes.ArticleName} into the seed extractor but it is full!",
+					$"{interaction.Performer.name} tries to place the {foodAtributes.ArticleName} into the seed extractor but it is full!");
+				return;
+			}
+
+			Chat.AddActionMsgToChat(interaction.Performer,
 				$"You place the {foodAtributes.ArticleName} into the seed extractor",
 				$"{interaction.Performer.name} places the {foodAtributes.name} into the seed extractor");
-				if (foodToBeProcessed.Count == 0 && currentState != PowerState.Off)
-				{
-					Chat.AddLocalMsgToChat("The seed extractor begins processing", gameObject);
-				}
-				foodToBeProcessed.Enqueue(grownFood);
-				return;
+			if (foodToBeProcessed.Count == 0 && currentState != PowerState.Off)
+			{
+				Chat.AddLocalMsgToChat("The seed extractor begins processing", gameObject);
 			}
-			// If no interaction happens
 			networkTab.ServerPerformInteraction(interaction);
+			foodToBeProcessed.Enqueue(grownFood);
+		}
+
+		private void AddSeedPacketToStorage(SeedPacket packet, HandApply interaction)
+		{
+			if (Inventory.ServerTransfer(interaction.HandSlot, storage.GetBestSlotFor(interaction.HandObject)))
+			{
+				seedPackets.Add(packet);
+				UpdateEvent.Invoke();
+				Chat.AddActionMsgToChat(interaction.Performer,
+					$"You place the {packet.gameObject.ExpensiveName()} into the seed extractor.",
+					$"{interaction.Performer.name} places the {packet.gameObject.ExpensiveName()} into the seed extractor.");
+				return;
+			}
+			Chat.AddActionMsgToChat(interaction.Performer,
+				$"You try and place the {packet.gameObject.ExpensiveName()} into the seed extractor but it is full!",
+				$"{interaction.Performer.name} tries to place the {packet.gameObject.ExpensiveName()} into the seed extractor but it is full!");
 		}
 
 		public bool WillInteract(HandApply interaction, NetworkSide side)
 		{
+			if (interaction.HandObject == null) return false;
 			return DefaultWillInteract.Default(interaction, side) &&
 			   interaction.TargetObject == gameObject &&
-			   interaction.HandObject != null &&
-			   interaction.HandObject.TryGetComponent<GrownFood>(out _);
+			   interaction.HandObject.TryGetComponent<GrownFood>(out _) || interaction.HandObject.TryGetComponent<SeedPacket>(out _);
```

Comment:
```suggestion
			   (interaction.HandObject.TryGetComponent<GrownFood>(out _) || interaction.HandObject.TryGetComponent<SeedPacket>(out _));
```

Think this needs to be bracket'd?

Reply:
done 👍